--- a/configure
+++ b/configure
@@ -109,6 +109,7 @@
 }
 
 my $prefix = '/usr/local/openresty';
+my $ngx_prefix;
 my %resty_opts;
 my $dry_run;
 my @ngx_rpaths;
@@ -145,6 +146,9 @@
     if ($opt =~ /^--prefix=(.*)/) {
         $prefix = $1;
 
+    } elsif ($opt =~ /^--ngx_prefix=(.*)/) {
+        $ngx_prefix = $1;
+
     } elsif ($opt eq '--without-lua51') {
         $resty_opts{no_lua} = 1;
 
@@ -253,7 +257,7 @@
 
 print "platform: $platform ($OS)\n";
 
-my $ngx_prefix = "$prefix/nginx";
+$ngx_prefix = "$prefix/nginx" unless $ngx_prefix;
 
 my $postamble = '';
 
